### PR TITLE
Issue #2706059 by Dave Reid: Filter out empty attributes.

### PR DESCRIPTION
**Pull request by davereid**
https://www.drupal.org/node/2706059

### CODE REVIEW

File: src/Form/EntityEmbedDialog.php [hunk 1]
```diff
@@ -737,6 +736,11 @@ public function submitEmbedStep(array &$form, FormStateInterface $form_state) {
         $values['attributes']['data-entity-embed-settings'] = Json::encode($values['attributes']['data-entity-embed-settings']);
       }
 
+      // Filter out empty attributes.
+      $values['attributes'] = array_filter($values['attributes'], function($value) {
```
Comment (slashrsm) [inline src/Form/EntityEmbedDialog.php:740]: array_filter($values['attributes']) won't do?

Comment (davereid) [inline src/Form/EntityEmbedDialog.php:740]: It would filter out an alt text or caption value of "0", so I figured strlen was safer in this case.

Comment (slashrsm) [inline src/Form/EntityEmbedDialog.php:740]: Oh joys of PHP! I guess "0" isn't the most relevant value, but I understand.

**Comment by slashrsm** [comment]
Merged rebased branch. Rebased commit is: 6cff3f3

**Pull request closed without merging**